### ISSUE TITLE
include issue assignment into transition_issue

### ISSUE THREAD

**Issue by pabgan** [pycontribs/jira#1588]
### Problem trying to solve

Transitioning an issue usually comes hand to hand with a change in the assignee like:

- Transition to **Code Review** and assign **dev2**.
- Transition to **Resolve** and assign **QA**.
- Transition to **Implementation** and assign to **me**.

Currently `transition_issue` provides no obvious way to do this. There might be a workaround using `fields` parameter but I could not manage to make it work and anyway, I think this one deserves first class support.

### Possible solution(s)

Change `transition_issue` definition to:

```
transition_issue(issue: str, transition: str, assignee: Optional[str] = None, fields: Optional[Dict[str, Any]] = None, comment: Optional[str] = None, worklog: Optional[str] = None, **fieldargs)
```

and make the change atomic with the transition, the comment and all the other changes.

### Alternatives

Include somewhere some tip with the workaround.

**Comment by dmcbride64** [comment]
Yes, it can be done with the fields parameter. See the example here:  https://jira.readthedocs.io/examples.html#transitions

Note that "assignee" needs to be on the same screen as the transitions, otherwise you will get an error.

**Comment by pabgan** [comment]
Thanks @dmcbride64 .

Is that new? I can not believe I did not see it before.

I am trying it. I keep getting 

```
        response text = {"errorMessages":[],"errors":{"assignee":"Issues must be assigned."}}
```

is that the error you are talking about?

I do not know what _"to be on the same screen"_ means, but I am in a hurry and have not investigated it; I just did not want to let another day slip without getting back to you.

**Comment by dmcbride64** [comment]
@pabgan 

See this explanation of screens:  https://confluence.atlassian.com/adminjiraserver/defining-a-screen-938847288.html
